dev/devicelab/bin/tasks/run_machine_concurrent_hot_reload.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:vm_service_client/vm_service_client.dart';
+
+import 'package:flutter_devicelab/framework/adb.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+void main() {
+  Map<String, dynamic> parseFlutterResponse(String line) {
+    if (line.startsWith('[') && line.endsWith(']')) {
+      try {
+        return JSON.decode(line)[0];
+      } catch (e) {
+      // Not valid JSON, so likely some other output that was surrounded by [brackets]
+        return null;
+      }
+    }
+    return null;
+  }
+  
+  task(() async {
+    int vmServicePort;
+    String appId;
+
+    final Device device = await devices.workingDevice;
+    await device.unlock();
+    final Directory appDir = dir(path.join(flutterDirectory.path, 'dev/integration_tests/ui'));
+    await inDirectory(appDir, () async {
+      final Completer<Null> ready = new Completer<Null>();
+      bool ok;
+      print('run: starting...');
+      final Process run = await startProcess(
+        path.join(flutterDirectory.path, 'bin', 'flutter'),
+        <String>['run', '--machine', '--verbose', '-d', device.deviceId, 'lib/commands.dart'],
+      );
+      final StreamController<String> stdout = new StreamController<String>.broadcast();
+      run.stdout
```

Comment:
We're doing this twice in this file (and lots of places in other files); perhaps in-line a small `Stream<List<int>>` ==> `Stream<String` function?

```
Stream<String> transformToLines(Stream<List<int>> byteStream) {
  return byteStream.transform(UTF8.decoder).transform(const LineSplitter());
}
```

Reply:
In 859a279 I made a function at the top, though not sure if you want more globally?

This is one of those places I wish Dart had extension methods... `stream.transformToLines()` versus `transformToLines(stream)`!